### DISCUSSION TITLE
error: Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '0')

### DISCUSSION THREAD

**Discussion by Ice-Hazymoon** [mikro-orm/mikro-orm#5163] [Q&A] [answered]
```ts
export const orm = await MikroORM.init({
    clientUrl: PGConfig.PG_DB_URL,
    entities: [
        User,
    ],
    debug: true,
});
```

```ts
@Entity({
    tableName: 'users',
}) 
export class User {
    @PrimaryKey({ type: 'int' })
    id!: number;

    @Property({ type: 'int',  nullable: true })
    inviter_id!: number | null;

    @ManyToOne(() => User, { nullable: true })
    inviter!: Ref<User> | null;

    @OneToMany(() => User, user => user.inviter, { lazy: true })
    invitedUsers = new Collection<User>(this);
}
```

I created an Entity like this, but an error occurred during the run

```
error: Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '0')
```

I think it is a problem with the `invitedUsers` part, but I don’t know how to solve it. I am a novice in ORM and I don’t know what caused it. I hope to get help, thank you!

**Comment by B4nan** [comment]
> error: Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '0')

You need to share at least the full stack trace, this is not very helpful on its own. Also, entity definitions don't throw, you should provide the code on what you are actually doing with the entities, that is what throws.

>     @OneToMany(() => User, user => user.inviter, { lazy: true })

Drop the `lazy` option, that's surely wrong. Relations are lazy by default, and `lazy` is only for scalars and owning sides (m:1 and 1:1 relations). Let me know if it was about that, we should handle this gracefully instead of failing with TypeError, but my point is `lazy: true` here won't have any effect.

>    @ManyToOne(() => User, { nullable: true })
>    inviter!: Ref<User> | null;

You are missing `ref: true` in the options (unless you use ts-morph).

**Reply by Ice-Hazymoon** [reply]
I tried modifying the code according to your suggestions, but the same error still occurs.

```
[info] MikroORM version: 6.0.5
[discovery] ORM entity discovery started, using ReflectMetadataProvider
[discovery] - processing entity User
error: Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '0')
    at file:///workspaces/supabase/functions/test/node_modules/.deno/@mikro-orm+core@6.0.5/node_modules/@mikro-orm/core/metadata/MetadataValidator.js:193:44
    at Array.filter (<anonymous>)
    at file:///workspaces/supabase/functions/test/node_modules/.deno/@mikro-orm+core@6.0.5/node_modules/@mikro-orm/core/metadata/MetadataValidator.js:189:22
    at Array.flatMap (<anonymous>)
    at MetadataValidator.validateDuplicateFieldNames (file:///workspaces/supabase/functions/test/node_modules/.deno/@mikro-orm+core@6.0.5/node_modules/@mikro-orm/core/metadata/MetadataValidator.js:187:38)
    at MetadataValidator.validateEntityDefinition (file:///workspaces/supabase/functions/test/node_modules/.deno/@mikro-orm+core@6.0.5/node_modules/@mikro-orm/core/metadata/MetadataValidator.js:39:14)
    at MetadataDiscovery.processEntity (file:///workspaces/supabase/functions/test/node_modules/.deno/@mikro-orm+core@6.0.5/node_modules/@mikro-orm/core/metadata/MetadataDiscovery.js:451:24)
    at MetadataDiscovery.processDiscoveredEntities (file:///workspaces/supabase/functions/test/node_modules/.deno/@mikro-orm+core@6.0.5/node_modules/@mikro-orm/core/metadata/MetadataDiscovery.js:112:37)
    at MetadataDiscovery.discover (file:///workspaces/supabase/functions/test/node_modules/.deno/@mikro-orm+core@6.0.5/node_modules/@mikro-orm/core/metadata/MetadataDiscovery.js:47:14)
    at eventLoopTick (ext:core/01_core.js:182:7)
```

```ts
import { Entity, Property, ManyToOne, OneToMany, PrimaryKey, Collection } from "@mikro-orm/core";
import { MikroORM, Ref } from '@mikro-orm/postgresql';
import { PGConfig } from "~/config/index.ts";

@Entity({
    tableName: 'users',
})
export class User {
    @PrimaryKey({ type: 'int' })
    id!: number;

    @Property({ type: 'int',  nullable: true})
    inviter_id!: number | null;

    @ManyToOne(() => User, { nullable: true })
    inviter!: Ref<User> | null;

    @OneToMany(() => User, user => user.inviter, { lazy: true })
    invitedUsers = new Collection<User>(this);
}

const orm = await MikroORM.init({
    clientUrl: PGConfig.PG_DB_URL,
    entities: [
        User,
    ],
    debug: true,
});
```

This is all the code, it appears that the issue is caused by something at [MetadataValidator.ts#L225](https://github.com/mikro-orm/mikro-orm/blob/c6df1ba4a12981d2d6268552efcc898053782dc8/packages/core/src/metadata/MetadataValidator.ts#L225). I printed the p variable, and the contents are as follows.

```js
{
  name: "invitedUsers",
  kind: "1:m",
  cascade: [ "persist" ],
  lazy: false,
  ref: false,
  entity: [Function (anonymous)],
  mappedBy: "inviter",
  type: "User",
  runtimeType: "unknown",
  unsigned: undefined,
  length: undefined,
  precision: undefined,
  scale: undefined,
  columnTypes: [ "int" ],
  customType: undefined
}
```

My runtime environment is Deno, which has good support for npm packages. I'm not sure if the issue is with Deno.

**Reply by B4nan** [reply, marked as answer]
> My runtime environment is Deno

Maybe mention such things right ahead :] Deno is not (officially) supported, I know some people tried and failed (see #3079). Maybe try disabling the validation via `checkDuplicateFieldNames` to see if it works fine otherwise.

**Reply by Ice-Hazymoon** [reply]
Thank you, this has effectively solved my problem. and it works well in Deno.

**Reply by B4nan** [reply]
Interesting, thanks, this should help with the validation error 01d8c261b9a92e9e0b067e2f38875ac5c419bf86
